sdk/communication/azure-communication-chat/src/main/java/com/azure/android/communication/chat/implementation/notifications/fcm/RegistrationKeyManager.java

```diff
@@ -0,0 +1,144 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+package com.azure.android.communication.chat.implementation.notifications.fcm;
+
+import android.util.Log;
+
+import com.azure.android.core.logging.ClientLogger;
+import com.azure.android.core.util.Base64Util;
+
+import java.io.File;
+import java.io.FileInputStream;
+import java.security.NoSuchAlgorithmException;
+import java.util.Stack;
+
+import javax.crypto.KeyGenerator;
+import javax.crypto.SecretKey;
+
+/**
+ * A singleton wrapper which manage the persistence and rotation of secrete keys and execution result related to #{RenewalTokenWorker}.
+ * The secrete keys are persisted and loaded using key-store API. We generate new secrete keys every refresh and evict expired keys
+ * from key-store every refresh.
+ *
+ * RegistrationKeyStore is used as object to manage the keys in file/memory.
+ */
+public final class RegistrationKeyManager {
+    private static RegistrationKeyManager registrationKeyManager;
+
+    private KeyGenerator keyGenerator;
+
+    private RegistrationKeyStore registrationKeyStore;
+
+    //The duration we persist keys in key-store
+    public static final int EXPIRATION_TIME_MINUTES = 45;
+
+    private ClientLogger clientLogger = new ClientLogger(RegistrationKeyManager.class);
+
+    private RegistrationKeyManager()  {
+        try {
+            this.keyGenerator = KeyGenerator.getInstance("AES");
+        } catch (NoSuchAlgorithmException e) {
+            throw new RuntimeException("KeyGenerator failed: " + e.getMessage());
+        }
+        this.keyGenerator.init(256);
+        registrationKeyStore = new RegistrationKeyStore();
+    }
+
+    public static RegistrationKeyManager instance() {
+        if (registrationKeyManager == null) {
+            synchronized (RegistrationKeyManager.class) {
+                if (registrationKeyManager == null) {
+                    registrationKeyManager = new RegistrationKeyManager();
+                }
+            }
+        }
+        return registrationKeyManager;
+    }
+
+    private long getInsertionTime(RegistrationKeyStore.RegistrationKeyEntry entry) {
+        return entry.getCreationTime();
+    }
+
+    private int getNumOfPairs() {
+        int numKeyPairs = 0;
+        try {
+            numKeyPairs = registrationKeyStore.getSize() / 2;
+        } catch (Exception e) {
+            clientLogger.logExceptionAsError(new RuntimeException("Failed to get size from key store", e));
+        }
+        Log.v("RegistrationContainer", "Number of pairs in key-store: " + numKeyPairs);
```

Comment:
Log using `ClientLogger`.

Reply:
I am not sure of the reason. But log using client logger does not log the message in the "run tab". I am adding log here mainly to help debugging.

Reply:
I'll look into why it doesn't show in the main console, but it definitely shows in LogCat.